src/Native/FactorizationMachineNative/FactorizationMachineCore.cpp

```diff
@@ -0,0 +1,175 @@
+#include <cmath>
+#include <cstring>
+#include <limits>
+#include <pmmintrin.h>
+#include "../Stdafx.h"
+
+// Compute the output value of the field-aware factorization, as the sum of the linear part and the latent part. 
+// The linear part is the inner product of linearWeights and featureValues.
+// The latent part is the sum of all intra-field interactions in one field f, for all fields possible. 
+EXPORT_API(void) CalculateIntermediateVariablesNative(int fieldCount, int latentDim, int count, _In_ int * fieldIndices, _In_ int * featureIndices, _In_ float * featureValues,
+    _In_ float * linearWeights, _In_ float * latentWeights, _Inout_ float * latentSum, _Out_ float * response)
+{
+    // The number of all possible fields
```

Comment:
do we do any input validation here?  At least asserts would be nice.

Reply:
The only one possible check is checking the signs of some parameters and indices (i.e., m, d, c, pf's elements, pi's elements). Note that adding checks for Release mode is not recommended because they two functions are highly optimized and only computation-oriented.